<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Demo for leukemia patient dataset from Golub et al @ supraHex 1.7.2</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">supraHex 1.7.2</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

<script>
  (function() {
    var cx = '012094530736424854714:5dmiiel8qeq';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Demo for leukemia patient dataset from Golub et al</h2>

Notes:
<li> All results are based on supraHex (version 1.7.2). </li>
<li> R scripts (i.e. R expressions) plus necessary comments are highlighted in light cyan, and the rest are outputs in the screen. </li>
<li> Images displayed below may be distorted, but should be normal in your screen. </li>
<li> Functions contained in supraHex 1.7.2 are hyperlinked in-place and also listed on the right side. </li>

<div class="row">
    <div class="span8">
      <pre>
      <div class='input'># This is a demo for leukemia patient dataset from Golub et al
# 
# This leukemia patient expression dataset (the learning set, available from <a href="http://www.ncbi.nlm.nih.gov/pubmed/10521349" target="10521349">http://www.ncbi.nlm.nih.gov/pubmed/10521349</a>) contains an expression matrix of 3,051 genes X 38 samples, involving two types of leukemia: 11 acute myeloid leukemia (AML) and 27 acute lymphoblastic leukemia (ALL). These 27 ALL are further subtyped into 19 B-cell ALL (ALL_B) and 8 T-cell ALL (ALL_T).
###############################################################################

# (I) Load the package and import data
library(supraHex)
data(Golub)
data <- Golub # a matrix of 3,051 genes expressed in 38 samples

# (II) Train the supra-hexagonal map with input data only
sMap <- <a href='sPipeline.html' target=sPipeline>sPipeline</a>(data)
</div>
<strong class='message'>Start at 2015-07-16 20:23:27</strong>
<strong class='message'></strong>
<strong class='message'>First, define topology of a map grid (2015-07-16 20:23:27)...</strong>
<strong class='message'>Second, initialise the codebook matrix (331 X 38) using 'linear' initialisation, given a topology and input data (2015-07-16 20:23:27)...</strong>
<strong class='message'>Third, get training at the rough stage (2015-07-16 20:23:27)...</strong>
<strong class='message'>	1 out of 2 (2015-07-16 20:23:27)</strong>
<strong class='message'>	updated (2015-07-16 20:23:27)</strong>
<strong class='message'>	2 out of 2 (2015-07-16 20:23:27)</strong>
<strong class='message'>	updated (2015-07-16 20:23:27)</strong>
<strong class='message'>Fourth, get training at the finetune stage (2015-07-16 20:23:27)...</strong>
<strong class='message'>	1 out of 5 (2015-07-16 20:23:27)</strong>
<strong class='message'>	updated (2015-07-16 20:23:28)</strong>
<strong class='message'>	2 out of 5 (2015-07-16 20:23:28)</strong>
<strong class='message'>	updated (2015-07-16 20:23:28)</strong>
<strong class='message'>	3 out of 5 (2015-07-16 20:23:28)</strong>
<strong class='message'>	updated (2015-07-16 20:23:28)</strong>
<strong class='message'>	4 out of 5 (2015-07-16 20:23:28)</strong>
<strong class='message'>	updated (2015-07-16 20:23:28)</strong>
<strong class='message'>	5 out of 5 (2015-07-16 20:23:28)</strong>
<strong class='message'>	updated (2015-07-16 20:23:28)</strong>
<strong class='message'>Next, identify the best-matching hexagon/rectangle for the input data (2015-07-16 20:23:28)...</strong>
<strong class='message'>Finally, append the response data (hits and mqe) into the sMap object (2015-07-16 20:23:29)...</strong>
<strong class='message'></strong>
<strong class='message'>Below are the summaries of the training results:</strong>
<strong class='message'>   dimension of input data: 3051x38
   xy-dimension of map grid: xdim=21, ydim=21
   grid lattice: hexa
   grid shape: suprahex
   dimension of grid coord: 331x2
   initialisation method: linear
   dimension of codebook matrix: 331x38
   mean quantization error: 23.6263000610265
</strong>
<strong class='message'>Below are the details of trainology:</strong>
<strong class='message'>   training algorithm: batch
   alpha type: invert
   training neighborhood kernel: gaussian
   trainlength (x input data length): 2 at rough stage; 5 at finetune stage
   radius (at rough stage): from 6 to 1.5
   radius (at finetune stage): from 1.5 to 1
</strong>
<strong class='message'>End at 2015-07-16 20:23:29</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'><a href='visHexMulComp.html' target=visHexMulComp>visHexMulComp</a>(sMap,title.rotate=10,colormap="darkgreen-lightgreen-lightpink-darkred")
<a href='sWriteData.html' target=sWriteData>sWriteData</a>(sMap, data, filename="Output_Golub.txt")
</div>
<a href="javascript:newWin('Golub-32.png', 'Golub-32.png', '1200', '857.142857142857')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Golub-32.png', width="400" /></a>
<div class='input'>## As you have seen, a figure displays the multiple components of trained map in a sample-specific manner. You also see that a .txt file has been saved in your disk. The output file has 1st column for your input data ID (an integer; otherwise the row names of input data matrix), and 2nd column for the corresponding index of best-matching hexagons (i.e. gene clusters). You can also force the input data to be output; type ?<a href='sWriteData.html' target=sWriteData>sWriteData</a> for details.

# (III) Visualise the map, including built-in indexes, data hits/distributions, distance between map nodes, and codebook matrix
<a href='visHexMapping.html' target=visHexMapping>visHexMapping</a>(sMap, mappingType="indexes")
## As you have seen, the smaller hexagons in the supra-hexagonal map are indexed as follows: start from the center, and then expand circularly outwards, and for each circle increase in an anti-clock order.

<a href='visHexMapping.html' target=visHexMapping>visHexMapping</a>(sMap, mappingType="hits")
## As you have seen, the number represents how many input data vectors are hitting each hexagon, the size of which is proportional to the number of hits.

<a href='visHexMapping.html' target=visHexMapping>visHexMapping</a>(sMap, mappingType="dist")
## As you have seen, map distance tells how far each hexagon is away from its neighbors, and the size of each hexagon is proportional to this distance.

<a href='visHexPattern.html' target=visHexPattern>visHexPattern</a>(sMap, plotType="lines")
## As you have seen, line plot displays the patterns associated with the codebook matrix. If multple colors are given, the points are also plotted. When the pattern involves both positive and negative values, zero horizental line is also shown.

<a href='visHexPattern.html' target=visHexPattern>visHexPattern</a>(sMap, plotType="bars")
## As you have seen, bar plot displays the patterns associated with the codebook matrix. When the pattern involves both positive and negative values, the zero horizental line is in the middle of the hexagon; otherwise at the top of the hexagon for all negative values, and at the bottom for all positive values.

# (IV) Perform partitioning operation on the map to obtain continuous clusters (i.e. gene meta-clusters) as they are different from gene clusters in an individual map node
sBase <- <a href='sDmatCluster.html' target=sDmatCluster>sDmatCluster</a>(sMap)
</div>
<a href="javascript:newWin('Golub-34.png', 'Golub-34.png', '1200', '1050')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Golub-34.png', width="400" /></a>
<div class='input'><a href='visDmatCluster.html' target=visDmatCluster>visDmatCluster</a>(sMap, sBase)
<a href='sWriteData.html' target=sWriteData>sWriteData</a>(sMap, data, sBase, filename="Output_base_Golub.txt")
</div>
<a href="javascript:newWin('Golub-36.png', 'Golub-36.png', '1200', '1050')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Golub-36.png', width="400" /></a>
<div class='input'>## As you have seen, each cluster is filled with the same continuous color, and the cluster index is marked in the seed node. Although different clusters are coded using different colors (randomly generated), it is unavoidable to have very similar colors filling in neighbouring clusters. In other words, neighbouring clusters are visually indiscernible. In this confusing situation, you can rerun the command <a href='visDmatCluster.html' target=visDmatCluster>visDmatCluster</a>(sMap, sBase) until neighbouring clusters are indeed filled with very different colors. An output .txt file has been saved in your disk. This file has 1st column for your input data ID (an integer; otherwise the row names of input data matrix), and 2nd column for the corresponding index of best-matching hexagons (i.e. gene clusters), and 3rd column for the cluster bases (i.e. gene meta-clusters). You can also force the input data to be output; type ?sWriteData for details.

# (V) Reorder the sample-specific components of the map to delineate relationships between samples
sReorder <- <a href='sCompReorder.html' target=sCompReorder>sCompReorder</a>(data,metric="pearson") # see Figure 8
</div>
<strong class='message'>Start at 2015-07-16 20:23:59</strong>
<strong class='message'></strong>
<strong class='message'>First, define topology of a map grid (2015-07-16 20:23:59)...</strong>
<strong class='message'>Second, initialise the codebook matrix (117 X 38) using 'linear' initialisation, given a topology and input data (2015-07-16 20:23:59)...</strong>
<strong class='message'>Third, get training at the rough stage (2015-07-16 20:23:59)...</strong>
<strong class='message'>	1 out of 1178 (2015-07-16 20:23:59)</strong>
<strong class='message'>	118 out of 1178 (2015-07-16 20:23:59)</strong>
<strong class='message'>	236 out of 1178 (2015-07-16 20:23:59)</strong>
<strong class='message'>	354 out of 1178 (2015-07-16 20:23:59)</strong>
<strong class='message'>	472 out of 1178 (2015-07-16 20:23:59)</strong>
<strong class='message'>	590 out of 1178 (2015-07-16 20:23:59)</strong>
<strong class='message'>	708 out of 1178 (2015-07-16 20:23:59)</strong>
<strong class='message'>	826 out of 1178 (2015-07-16 20:23:59)</strong>
<strong class='message'>	944 out of 1178 (2015-07-16 20:24:00)</strong>
<strong class='message'>	1062 out of 1178 (2015-07-16 20:24:00)</strong>
<strong class='message'>	1178 out of 1178 (2015-07-16 20:24:00)</strong>
<strong class='message'>Fourth, get training at the finetune stage (2015-07-16 20:24:00)...</strong>
<strong class='message'>	1 out of 4712 (2015-07-16 20:24:00)</strong>
<strong class='message'>	472 out of 4712 (2015-07-16 20:24:00)</strong>
<strong class='message'>	944 out of 4712 (2015-07-16 20:24:00)</strong>
<strong class='message'>	1416 out of 4712 (2015-07-16 20:24:01)</strong>
<strong class='message'>	1888 out of 4712 (2015-07-16 20:24:01)</strong>
<strong class='message'>	2360 out of 4712 (2015-07-16 20:24:01)</strong>
<strong class='message'>	2832 out of 4712 (2015-07-16 20:24:01)</strong>
<strong class='message'>	3304 out of 4712 (2015-07-16 20:24:02)</strong>
<strong class='message'>	3776 out of 4712 (2015-07-16 20:24:02)</strong>
<strong class='message'>	4248 out of 4712 (2015-07-16 20:24:02)</strong>
<strong class='message'>	4712 out of 4712 (2015-07-16 20:24:03)</strong>
<strong class='message'>Next, identify the best-matching hexagon/rectangle for the input data (2015-07-16 20:24:03)...</strong>
<strong class='message'>Finally, append the response data (hits and mqe) into the sMap object (2015-07-16 20:24:03)...</strong>
<strong class='message'></strong>
<strong class='message'>Below are the summaries of the training results:</strong>
<strong class='message'>   dimension of input data: 38x38
   xy-dimension of map grid: xdim=13, ydim=9
   grid lattice: rect
   grid shape: sheet
   dimension of grid coord: 117x2
   initialisation method: linear
   dimension of codebook matrix: 117x38
   mean quantization error: 0.0621784272542971
</strong>
<strong class='message'>Below are the details of trainology:</strong>
<strong class='message'>   training algorithm: sequential
   alpha type: invert
   training neighborhood kernel: gaussian
   trainlength (x input data length): 31 at rough stage; 124 at finetune stage
   radius (at rough stage): from 2 to 1
   radius (at finetune stage): from 1 to 1
</strong>
<strong class='message'>End at 2015-07-16 20:24:03</strong>
<strong class='message'>Runtime in total is: 4 secs
</strong>
<div class='input'><a href='visCompReorder.html' target=visCompReorder>visCompReorder</a>(sMap,sReorder,title.rotate=15,colormap="darkgreen-lightgreen-lightpink-darkred")
## As you have seen, reordered components of trained map is displayed. Each component illustrates a sample-specific map and is placed within a two-dimensional rectangular lattice. Across components/samples, genes with similar expression patterns are mapped onto the same position of the map. Geometric locations of components delineate relationships between components/samples, that is, samples with the similar expression profiles are placed closer to each other.

# (VI) Build and visualise the bootstrapped tree
D <- t(data)
</div>
<a href="javascript:newWin('Golub-76.png', 'Golub-76.png', '1200', '1050')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Golub-76.png', width="400" /></a>
<div class='input'>rownames(D) <- paste(rownames(D), 1:nrow(D), sep=".") # temporally make sure the row names are unique
tree_bs <- <a href='visTreeBootstrap.html' target=visTreeBootstrap>visTreeBootstrap</a>(D, nodelabels.arg=list(cex=0.7))
</div>
<strong class='message'>Start at 2015-07-16 20:24:05</strong>
<strong class='message'></strong>
<strong class='message'>First, build the tree (using nj algorithm and euclidean distance) from input matrix (38 by 3051)...</strong>
<strong class='message'>Second, perform bootstrap analysis with 100 replicates...</strong>
<strong class='message'>Finally, visualise the bootstrapped tree...</strong>
<strong class='message'></strong>
<strong class='message'>Finish at 2015-07-16 20:24:06</strong>
<strong class='message'>Runtime in total is: 1 secs
</strong>
<a href="javascript:newWin('Golub-86.png', 'Golub-86.png', '1200', '1050')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Golub-86.png', width="400" /></a>
<div class='input'>## As you have seen, neighbour-joining tree is constructed based on pairwise euclidean distance matrices between samples. The robustness of tree branching is evaluated using bootstraping. In internal nodes (also color-coded), the number represents the proportion of bootstrapped trees that support the observed internal branching. The higher the number, the more robust the tree branching. 100 means that the internal branching is always observed by resampling characters/genes. 

# (VII) Visualise the matrix using heatmap
# The samples are ordered according to the neighbour-joining tree
flag <- match(tree_bs$tip.label, rownames(D))
rownames(D) <- sub("\\.\\d+$", "", rownames(D)) # restore the original names
D <- D[flag,]
# prepare colors for the column sidebar of heatmap
# color for AML/ALL types
types <- sub("_.*","",rownames(D))
lvs <- unique(types)
lvs_color <- <a href='visColormap.html' target=visColormap>visColormap</a>(colormap="darkblue-darkorange")(length(lvs))
col_types <- sapply(types, function(x) lvs_color[x==lvs])
# color for ALL subtypes
subtypes <- sub(".*_","",rownames(D))
lvs <- unique(subtypes)
lvs_color <- <a href='visColormap.html' target=visColormap>visColormap</a>(colormap="gray-black")(length(lvs))
col_subtypes <- sapply(subtypes, function(x) lvs_color[x==lvs])
# combine both color vectors
ColSideColors <- cbind(col_subtypes,col_types)
colnames(ColSideColors) <- c("ALL subtypes", "AML/ALL types")
# heatmap embeded with sidebars annotating samples
<a href='visHeatmapAdv.html' target=visHeatmapAdv>visHeatmapAdv</a>(t(D), Rowv=T, Colv=F, dendrogram="none", colormap="darkgreen-lightgreen-lightpink-darkred", ColSideColors=ColSideColors, ColSideHeight=0.4, ColSideLabelLocation="left", labRow=NA)</div>
<a href="javascript:newWin('Golub-88.png', 'Golub-88.png', '1200', '1050')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Golub-88.png', width="400" /></a>
      </pre>
    </div>

  <div class="span4">

    <h2>Source demo</h2>
    <a href="javascript:newWin('demo/Golub.r', 'demo/Golub.r', '1200', '1200')" title="Click to view"><B><code>Golub.r</code></B></a>
    
    <h2>Functions used in this demo</h2>
    <ul>
      <li><a href="sPipeline.html">sPipeline</a></li>
      <li><a href="visHexMulComp.html">visHexMulComp</a></li>
      <li><a href="sWriteData.html">sWriteData</a></li>
      <li><a href="visHexMapping.html">visHexMapping</a></li>
      <li><a href="visHexPattern.html">visHexPattern</a></li>
      <li><a href="sDmatCluster.html">sDmatCluster</a></li>
      <li><a href="visDmatCluster.html">visDmatCluster</a></li>
      <li><a href="sCompReorder.html">sCompReorder</a></li>
      <li><a href="visCompReorder.html">visCompReorder</a></li>
      <li><a href="visTreeBootstrap.html">visTreeBootstrap</a></li>
      <li><a href="visColormap.html">visColormap</a></li>
      <li><a href="visHeatmapAdv.html">visHeatmapAdv</a></li>
    </ul>
    
    <!--
    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) supraHex: an R/Bioconductor package for tabular omics data analysis using a supra-hexagonal map. <I>Biochemical and Biophysical Research Communications</I>, 443(1), 285-289. <a href="http://dx.doi.org/10.1016/j.bbrc.2013.11.103">doi:10.1016/j.bbrc.2013.11.103</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24309102">24309102</a></p>

    

    
    <h2>URL</h2>
    <ul>
      <li><a href="http://suprahex.r-forge.r-project.org">http://suprahex.r-forge.r-project.org</a></li>
    </ul>
    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    -->
    
  </div>

</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@well.ox.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>