<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Demo for for the artwork in ISMB 2014 @ supraHex 1.7.1</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">supraHex 1.7.1</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

<script>
  (function() {
    var cx = '012094530736424854714:5dmiiel8qeq';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Demo for for the artwork in ISMB 2014</h2>

Notes:
<li> All results are based on supraHex (version 1.7.1). </li>
<li> R scripts (i.e. R expressions) plus necessary comments are highlighted in light cyan, and the rest are outputs in the screen. </li>
<li> Images displayed below may be distorted, but should be normal in your screen. </li>
<li> Functions contained in supraHex 1.7.1 are hyperlinked in-place and also listed on the right side. </li>

<div class="row">
    <div class="span8">
      <pre>
      <div class='input'># This is a demo for the artwork in ISMB 2014
# 
# This artwork is produced using the human embryo expression dataset (available from <a href="http://www.ncbi.nlm.nih.gov/pubmed/20643359" target="20643359">http://www.ncbi.nlm.nih.gov/pubmed/20643359</a>) involves six successive developmental stages (S9-S14) with three replicates (R1-R3) for each stage, including:
## Fang: an expression matrix of 5,441 genes X 18 samples;
## Fang.geneinfo: a matrix of 5,441 X 3 containing gene information;
## Fang.sampleinfo: a matrix of 18 X 3 containing sample information.
###############################################################################

# Load or install packages used in this demo
for(pkg in c("hexbin","ape","supraHex","graph","Rgraphviz","igraph","Biobase","limma","dnet")){
    if(!require(pkg, character.only=T)){
        source("http://bioconductor.org/biocLite.R")
        biocLite(pkg)
        lapply(pkg, library, character.only=T)
    }
}
</div>
<strong class='message'>Loading required package: ape</strong>
<strong class='message'>Loading required package: graph</strong>
<strong class='message'>
Attaching package: ‘graph’
</strong>
<strong class='message'>The following object is masked from ‘package:ape’:

    edges
</strong>
<strong class='message'>Loading required package: Rgraphviz</strong>
<strong class='message'>Loading required package: igraph</strong>
<strong class='message'>
Attaching package: ‘igraph’
</strong>
<strong class='message'>The following objects are masked from ‘package:graph’:

    degree, edges
</strong>
<strong class='message'>The following object is masked from ‘package:ape’:

    edges
</strong>
<strong class='message'>Loading required package: Biobase</strong>
<strong class='message'>Loading required package: BiocGenerics</strong>
<strong class='message'>Loading required package: parallel</strong>
<strong class='message'>
Attaching package: ‘BiocGenerics’
</strong>
<strong class='message'>The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB
</strong>
<strong class='message'>The following object is masked from ‘package:stats’:

    xtabs
</strong>
<strong class='message'>The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, as.vector, cbind, colnames,
    do.call, duplicated, eval, evalq, Filter, Find, get, intersect,
    is.unsorted, lapply, Map, mapply, match, mget, order, paste, pmax,
    pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce, rep.int,
    rownames, sapply, setdiff, sort, table, tapply, union, unique,
    unlist
</strong>
<strong class='message'>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.
</strong>
<strong class='message'>Loading required package: limma</strong>
<strong class='message'>
Attaching package: ‘limma’
</strong>
<strong class='message'>The following object is masked from ‘package:BiocGenerics’:

    plotMA
</strong>
<strong class='message'>Loading required package: dnet</strong>
<div class='input'>
############################################################
# Preprocess data (this should be done as your data suggest)
############################################################

# import data containing three variables ('Fang', 'Fang.geneinfo' and 'Fang.sampleinfo')
data(Fang)
data <- Fang
fdata <- as.data.frame(Fang.geneinfo[,2:3])
rownames(fdata) <- Fang.geneinfo[,1]
pdata <- as.data.frame(Fang.sampleinfo[,2:3])
rownames(pdata) <- Fang.sampleinfo[,1]

# create the 'eset' object
colmatch <- match(rownames(pdata),colnames(data))
rowmatch <- match(rownames(fdata),rownames(data))
data <- data[rowmatch,colmatch]
eset <- new("ExpressionSet", exprs=as.matrix(data), phenoData=as(pdata,"AnnotatedDataFrame"), featureData=as(fdata,"AnnotatedDataFrame"))

# A function to convert probeset-centric to entrezgene-centric expression levels
prob2gene <- function(eset){
    fdat <- fData(eset)
    tmp <- as.matrix(unique(fdat[c("EntrezGene", "Symbol")]))
    forder <- tmp[order(as.numeric(tmp[,1])),]
    forder <- forder[!is.na(forder[,1]),]
    rownames(forder) <- forder[,2]
    system.time({
        dat <- exprs(eset)
        edat <- matrix(data=NA, nrow=nrow(forder), ncol=ncol(dat))
        for (i in 1:nrow(forder)){
            ind <- which(fdat$EntrezGene==as.numeric(forder[i,1]))
            if (length(ind) == 1){
                edat[i,] <- dat[ind,]
            }else{
                edat[i,] <- apply(dat[ind,],2,mean)
            }
        }
    })
    
    rownames(edat) <- rownames(forder) # as gene symbols
    colnames(edat) <- rownames(pData(eset))
    esetGene <- new('ExpressionSet',exprs=data.frame(edat),phenoData=as(pData(eset),"AnnotatedDataFrame"),featureData=as(data.frame(forder),"AnnotatedDataFrame"))
    return(esetGene)
}
esetGene <- prob2gene(eset)
esetGene
</div>
<div class='output'>ExpressionSet (storageMode: lockedEnvironment)
assayData: 4139 features, 18 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: S9_R1 S9_R2 ... S14_R3 (18 total)
  varLabels: Stage Replicate
  varMetadata: labelDescription
featureData
  featureNames: A2M AADAC ... LOC100131613 (4139 total)
  fvarLabels: EntrezGene Symbol
  fvarMetadata: labelDescription
experimentData: use 'experimentData(object)'
Annotation:  
</div>
<div class='input'>
# prepare the expression matrix: relative to mean expression level across stages
D <- as.matrix(exprs(esetGene))
D <- D - as.matrix(apply(D,1,mean),ncol=1)[,rep(1,ncol(D))]

# only focus on differentially expressed genes
library(dnet)
## define the design matrix in a order manner
all <- as.vector(pData(esetGene)$Stage)
level <- levels(factor(all))
index_level <- sapply(level, function(x) which(all==x)[1])
level_sorted <- all[sort(index_level, decreasing=F)]
design <- sapply(level_sorted, function(x) as.numeric(all==x)) # Convert a factor column to multiple boolean columns
## a linear model is fitted for every gene by the function lmFit
fit <- lmFit(exprs(esetGene), design)
## define a contrast matrix
contrasts <- dContrast(level_sorted, contrast.type="average")
contrast.matrix <- makeContrasts(contrasts=contrasts$each, levels=design)
colnames(contrast.matrix) <- contrasts$name
## computes moderated t-statistics and log-odds of differential expression by empirical Bayes shrinkage of the standard errors towards a common value
fit2 <- contrasts.fit(fit, contrast.matrix)
fit2 <- eBayes(fit2)
## for p-value
pvals <- as.matrix(fit2$p.value)
## for adjusted p-value
adjpvals <- sapply(1:ncol(pvals),function(x) {
    p.adjust(pvals[,x], method="BH")
})
colnames(adjpvals) <- colnames(pvals)
## select differentially expressed genes
flag <- apply(adjpvals<1e-2, 1, sum)
data <- as.matrix(fit2$coefficients[flag>=1,])

############################################################
# Now analyse 'data' prepared above by the package 'supraHex'
############################################################
# (I) Train the supra-hexagonal map with input data only
sMap <- <a href='sPipeline.html' target=sPipeline>sPipeline</a>(data, xdim=13)
</div>
<strong class='message'>Start at 2015-06-04 10:43:41</strong>
<strong class='message'></strong>
<strong class='message'>First, define topology of a map grid (2015-06-04 10:43:41)...</strong>
<strong class='message'>Second, initialise the codebook matrix (127 X 6) using 'linear' initialisation, given a topology and input data (2015-06-04 10:43:41)...</strong>
<strong class='message'>Third, get training at the rough stage (2015-06-04 10:43:41)...</strong>
<strong class='message'>	1 out of 2 (2015-06-04 10:43:41)</strong>
<strong class='message'>	updated (2015-06-04 10:43:41)</strong>
<strong class='message'>	2 out of 2 (2015-06-04 10:43:41)</strong>
<strong class='message'>	updated (2015-06-04 10:43:41)</strong>
<strong class='message'>Fourth, get training at the finetune stage (2015-06-04 10:43:41)...</strong>
<strong class='message'>	1 out of 2 (2015-06-04 10:43:41)</strong>
<strong class='message'>	updated (2015-06-04 10:43:41)</strong>
<strong class='message'>	2 out of 2 (2015-06-04 10:43:41)</strong>
<strong class='message'>	updated (2015-06-04 10:43:41)</strong>
<strong class='message'>Next, identify the best-matching hexagon/rectangle for the input data (2015-06-04 10:43:41)...</strong>
<strong class='message'>Finally, append the response data (hits and mqe) into the sMap object (2015-06-04 10:43:41)...</strong>
<strong class='message'></strong>
<strong class='message'>Below are the summaries of the training results:</strong>
<strong class='message'>   dimension of input data: 4018x6
   xy-dimension of map grid: xdim=13, ydim=13
   grid lattice: hexa
   grid shape: suprahex
   dimension of grid coord: 127x2
   initialisation method: linear
   dimension of codebook matrix: 127x6
   mean quantization error: 0.108267518704114
</strong>
<strong class='message'>Below are the details of trainology:</strong>
<strong class='message'>   training algorithm: batch
   alpha type: invert
   training neighborhood kernel: gaussian
   trainlength (x input data length): 2 at rough stage; 2 at finetune stage
   radius (at rough stage): from 4 to 1
   radius (at finetune stage): from 1 to 1
</strong>
<strong class='message'>End at 2015-06-04 10:43:41</strong>
<strong class='message'>Runtime in total is: 0 secs
</strong>
<div class='input'>
# (II) Perform partitioning operation on the map to obtain continuous clusters (i.e. gene meta-clusters) as they are different from gene clusters in an individual map node
sBase <- <a href='sDmatCluster.html' target=sDmatCluster>sDmatCluster</a>(sMap, which_neigh=2, reindexSeed="svd")

# (III) Define visuals
## the hexagon frame according to mete-clusters
myColor <- c("transparent", "black")
border.color <- myColor[sBase$bases%%2 + 1]
## the size of hexagon according to distance to their neighours
dMat <- <a href='sDmat.html' target=sDmat>sDmat</a>(sMap, which_neigh = 2, distMeasure = "median")

# (IV) Do visualisation in a single map
<a href='visDmatCluster.html' target=visDmatCluster>visDmatCluster</a>(sMap,sBase, gp=grid::gpar(cex=1.5, font=2, col="transparent"), colormap="PapayaWhip-pink-Tomato", area.size=-1*log2(dMat), border.color=border.color)
</div>
<a href="javascript:newWin('ISMB2014-50.png', 'ISMB2014-50.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='ISMB2014-50.png', width="400" /></a>
<div class='input'># As you have seen, the map distance (the hexagon size) tells how far each node is away from its neighbors, thus characterising relationships between clustered genes. The map is also partitioned to obtain gene meta-clusters covering continuous regions, as colour-coded by the 'potato-peach-tomato' colormap.</div>
      </pre>
    </div>

  <div class="span4">

    <h2>Source demo</h2>
    <a href="javascript:newWin('demo/ISMB2014.r', 'demo/ISMB2014.r', '1200', '1200')" title="Click to view"><B><code>ISMB2014.r</code></B></a>
    
    <h2>Functions used in this demo</h2>
    <ul>
      <li><a href="sPipeline.html">sPipeline</a></li>
      <li><a href="sDmatCluster.html">sDmatCluster</a></li>
      <li><a href="sDmat.html">sDmat</a></li>
      <li><a href="visDmatCluster.html">visDmatCluster</a></li>
    </ul>
    
    <!--
    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) supraHex: an R/Bioconductor package for tabular omics data analysis using a supra-hexagonal map. <I>Biochemical and Biophysical Research Communications</I>, 443(1), 285-289. <a href="http://dx.doi.org/10.1016/j.bbrc.2013.11.103">doi:10.1016/j.bbrc.2013.11.103</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24309102">24309102</a></p>

    

    
    <h2>URL</h2>
    <ul>
      <li><a href="http://suprahex.r-forge.r-project.org">http://suprahex.r-forge.r-project.org</a></li>
    </ul>
    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    -->
    
  </div>

</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@well.ox.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>