<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Demo for human embryo dataset from Fang et al @ supraHex 1.7.2</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">supraHex 1.7.2</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

<script>
  (function() {
    var cx = '012094530736424854714:5dmiiel8qeq';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Demo for human embryo dataset from Fang et al</h2>

Notes:
<li> All results are based on supraHex (version 1.7.2). </li>
<li> R scripts (i.e. R expressions) plus necessary comments are highlighted in light cyan, and the rest are outputs in the screen. </li>
<li> Images displayed below may be distorted, but should be normal in your screen. </li>
<li> Functions contained in supraHex 1.7.2 are hyperlinked in-place and also listed on the right side. </li>

<div class="row">
    <div class="span8">
      <pre>
      <div class='input'># This is a demo for human embryo dataset from Fang et al
# 
# This human embryo expression dataset (available from <a href="http://www.ncbi.nlm.nih.gov/pubmed/20643359" target="20643359">http://www.ncbi.nlm.nih.gov/pubmed/20643359</a>) involves six successive developmental stages (S9-S14) with three replicates (R1-R3) for each stage, including:
## Fang: an expression matrix of 5,441 genes X 18 samples;
## Fang.geneinfo: a matrix of 5,441 X 3 containing gene information;
## Fang.sampleinfo: a matrix of 18 X 3 containing sample information.
###############################################################################

# (I) Load the package and import data
library(supraHex)
data(Fang) # import aforementioned three variables ('Fang', 'Fang.geneinfo' and 'Fang.sampleinfo')
# a matrix of 5,441 genes expressed in 18 samples
# transform data by row/gene centering
data <- Fang - matrix(rep(apply(Fang,1,mean),ncol(Fang)),ncol=ncol(Fang))

# (II) Train the supra-hexagonal map with input data only
sMap <- <a href='sPipeline.html' target=sPipeline>sPipeline</a>(data)
</div>
<strong class='message'>Start at 2015-07-16 20:24:15</strong>
<strong class='message'></strong>
<strong class='message'>First, define topology of a map grid (2015-07-16 20:24:15)...</strong>
<strong class='message'>Second, initialise the codebook matrix (397 X 18) using 'linear' initialisation, given a topology and input data (2015-07-16 20:24:15)...</strong>
<strong class='message'>Third, get training at the rough stage (2015-07-16 20:24:15)...</strong>
<strong class='message'>	1 out of 2 (2015-07-16 20:24:15)</strong>
<strong class='message'>	updated (2015-07-16 20:24:15)</strong>
<strong class='message'>	2 out of 2 (2015-07-16 20:24:15)</strong>
<strong class='message'>	updated (2015-07-16 20:24:15)</strong>
<strong class='message'>Fourth, get training at the finetune stage (2015-07-16 20:24:15)...</strong>
<strong class='message'>	1 out of 3 (2015-07-16 20:24:15)</strong>
<strong class='message'>	updated (2015-07-16 20:24:16)</strong>
<strong class='message'>	2 out of 3 (2015-07-16 20:24:16)</strong>
<strong class='message'>	updated (2015-07-16 20:24:16)</strong>
<strong class='message'>	3 out of 3 (2015-07-16 20:24:16)</strong>
<strong class='message'>	updated (2015-07-16 20:24:17)</strong>
<strong class='message'>Next, identify the best-matching hexagon/rectangle for the input data (2015-07-16 20:24:17)...</strong>
<strong class='message'>Finally, append the response data (hits and mqe) into the sMap object (2015-07-16 20:24:17)...</strong>
<strong class='message'></strong>
<strong class='message'>Below are the summaries of the training results:</strong>
<strong class='message'>   dimension of input data: 5441x18
   xy-dimension of map grid: xdim=23, ydim=23
   grid lattice: hexa
   grid shape: suprahex
   dimension of grid coord: 397x2
   initialisation method: linear
   dimension of codebook matrix: 397x18
   mean quantization error: 0.356873218409639
</strong>
<strong class='message'>Below are the details of trainology:</strong>
<strong class='message'>   training algorithm: batch
   alpha type: invert
   training neighborhood kernel: gaussian
   trainlength (x input data length): 2 at rough stage; 3 at finetune stage
   radius (at rough stage): from 6 to 1.5
   radius (at finetune stage): from 1.5 to 1
</strong>
<strong class='message'>End at 2015-07-16 20:24:17</strong>
<strong class='message'>Runtime in total is: 2 secs
</strong>
<div class='input'><a href='visHexMulComp.html' target=visHexMulComp>visHexMulComp</a>(sMap, title.rotate=10)
<a href='sWriteData.html' target=sWriteData>sWriteData</a>(sMap, data, filename="Output_Fang.txt")
</div>
<a href="javascript:newWin('Fang-28.png', 'Fang-28.png', '1200', '840')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-28.png', width="400" /></a>
<div class='input'>## As you have seen, a figure displays the multiple components of trained map in a sample-specific manner. You also see that a .txt file has been saved in your disk. The output file has 1st column for your input data ID (an integer; otherwise the row names of input data matrix), and 2nd column for the corresponding index of best-matching hexagons (i.e. gene clusters). You can also force the input data to be output; type ?<a href='sWriteData.html' target=sWriteData>sWriteData</a> for details.

# (III) Visualise the map, including built-in indexes, data hits/distributions, distance between map nodes, and codebook matrix
<a href='visHexMapping.html' target=visHexMapping>visHexMapping</a>(sMap, mappingType="indexes")
## As you have seen, the smaller hexagons in the supra-hexagonal map are indexed as follows: start from the center, and then expand circularly outwards, and for each circle increase in an anti-clock order.

<a href='visHexMapping.html' target=visHexMapping>visHexMapping</a>(sMap, mappingType="hits")
## As you have seen, the number represents how many input data vectors are hitting each hexagon, the size of which is proportional to the number of hits.

<a href='visHexMapping.html' target=visHexMapping>visHexMapping</a>(sMap, mappingType="dist")
## As you have seen, map distance tells how far each hexagon is away from its neighbors, and the size of each hexagon is proportional to this distance.

<a href='visHexPattern.html' target=visHexPattern>visHexPattern</a>(sMap, plotType="lines")
## As you have seen, line plot displays the patterns associated with the codebook matrix. If multple colors are given, the points are also plotted. When the pattern involves both positive and negative values, zero horizental line is also shown.

<a href='visHexPattern.html' target=visHexPattern>visHexPattern</a>(sMap, plotType="bars")
## As you have seen, bar plot displays the patterns associated with the codebook matrix. When the pattern involves both positive and negative values, the zero horizental line is in the middle of the hexagon; otherwise at the top of the hexagon for all negative values, and at the bottom for all positive values.

# (IV) Perform partitioning operation on the map to obtain continuous clusters (i.e. gene meta-clusters) as they are different from gene clusters in an individual map node
sBase <- <a href='sDmatCluster.html' target=sDmatCluster>sDmatCluster</a>(sMap)
</div>
<a href="javascript:newWin('Fang-30.png', 'Fang-30.png', '1200', '1000')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-30.png', width="400" /></a>
<div class='input'><a href='visDmatCluster.html' target=visDmatCluster>visDmatCluster</a>(sMap, sBase)
<a href='sWriteData.html' target=sWriteData>sWriteData</a>(sMap, data, sBase, filename="Output_base_Fang.txt")
</div>
<a href="javascript:newWin('Fang-32.png', 'Fang-32.png', '1200', '1000')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-32.png', width="400" /></a>
<div class='input'>## As you have seen, each cluster is filled with the same continuous color, and the cluster index is marked in the seed node. Although different clusters are coded using different colors (randomly generated), it is unavoidable to have very similar colors filling in neighbouring clusters. In other words, neighbouring clusters are visually indiscernible. In this confusing situation, you can rerun the command <a href='visDmatCluster.html' target=visDmatCluster>visDmatCluster</a>(sMap, sBase) until neighbouring clusters are indeed filled with very different colors. An output .txt file has been saved in your disk. This file has 1st column for your input data ID (an integer; otherwise the row names of input data matrix), and 2nd column for the corresponding index of best-matching hexagons (i.e. gene clusters), and 3rd column for the cluster bases (i.e. gene meta-clusters). You can also force the input data to be output; type ?sWriteData for details.

# prepare colors for the column sidebar
# color for stages (S9-S14)
stages <- sub("_.*","",colnames(data))
lvs <- unique(stages)
lvs_color <- <a href='visColormap.html' target=visColormap>visColormap</a>(colormap="jet")(length(lvs))
col_stages <- sapply(stages, function(x) lvs_color[x==lvs])
# color for replicates (R1-R3)
replicates <- sub(".*_","",colnames(data))
lvs <- unique(replicates)
lvs_color <- <a href='visColormap.html' target=visColormap>visColormap</a>(colormap="gray-black")(length(lvs))
col_replicates <- sapply(replicates, function(x) lvs_color[x==lvs])
# combine both color vectors
ColSideColors <- cbind(col_stages,col_replicates)
colnames(ColSideColors) <- c("Stages","Replicates")
output <- <a href='visDmatHeatmap.html' target=visDmatHeatmap>visDmatHeatmap</a>(sMap, data, sBase, base.legend.location="bottomleft", reorderRow="hclust", ColSideColors=ColSideColors, KeyValueName="log2(Ratio)", ColSideLabelLocation="right", labRow=NA)
</div>
<a href="javascript:newWin('Fang-34.png', 'Fang-34.png', '1200', '1000')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-34.png', width="400" /></a>
<div class='input'>## As you have seen, heatmap is used to visualise patterns seen in genes within each meta-cluster/base. Row side bar indicates the meta-clusters/bases. Column side bar annotates samples. The returned variable "output" (NOT a txt file) has 1st column for your input data ID (an integer; otherwise the row names of input data matrix), and 2nd column for the corresponding index of best-matching hexagons (i.e. gene clusters), and 3rd column for the cluster bases (i.e. gene meta-clusters). Note: it has rows in the same order as visualised in the heatmap

# (V) Reorder the sample-specific components of the map to delineate relationships between samples
sReorder <- <a href='sCompReorder.html' target=sCompReorder>sCompReorder</a>(data, metric="euclidean")
</div>
<strong class='message'>Start at 2015-07-16 20:24:59</strong>
<strong class='message'></strong>
<strong class='message'>First, define topology of a map grid (2015-07-16 20:24:59)...</strong>
<strong class='message'>Second, initialise the codebook matrix (54 X 18) using 'linear' initialisation, given a topology and input data (2015-07-16 20:24:59)...</strong>
<strong class='message'>Third, get training at the rough stage (2015-07-16 20:24:59)...</strong>
<strong class='message'>	1 out of 540 (2015-07-16 20:24:59)</strong>
<strong class='message'>	54 out of 540 (2015-07-16 20:24:59)</strong>
<strong class='message'>	108 out of 540 (2015-07-16 20:24:59)</strong>
<strong class='message'>	162 out of 540 (2015-07-16 20:24:59)</strong>
<strong class='message'>	216 out of 540 (2015-07-16 20:24:59)</strong>
<strong class='message'>	270 out of 540 (2015-07-16 20:24:59)</strong>
<strong class='message'>	324 out of 540 (2015-07-16 20:24:59)</strong>
<strong class='message'>	378 out of 540 (2015-07-16 20:24:59)</strong>
<strong class='message'>	432 out of 540 (2015-07-16 20:24:59)</strong>
<strong class='message'>	486 out of 540 (2015-07-16 20:24:59)</strong>
<strong class='message'>	540 out of 540 (2015-07-16 20:24:59)</strong>
<strong class='message'>Fourth, get training at the finetune stage (2015-07-16 20:24:59)...</strong>
<strong class='message'>	1 out of 2160 (2015-07-16 20:24:59)</strong>
<strong class='message'>	216 out of 2160 (2015-07-16 20:24:59)</strong>
<strong class='message'>	432 out of 2160 (2015-07-16 20:24:59)</strong>
<strong class='message'>	648 out of 2160 (2015-07-16 20:24:59)</strong>
<strong class='message'>	864 out of 2160 (2015-07-16 20:24:59)</strong>
<strong class='message'>	1080 out of 2160 (2015-07-16 20:24:59)</strong>
<strong class='message'>	1296 out of 2160 (2015-07-16 20:24:59)</strong>
<strong class='message'>	1512 out of 2160 (2015-07-16 20:24:59)</strong>
<strong class='message'>	1728 out of 2160 (2015-07-16 20:25:00)</strong>
<strong class='message'>	1944 out of 2160 (2015-07-16 20:25:00)</strong>
<strong class='message'>	2160 out of 2160 (2015-07-16 20:25:00)</strong>
<strong class='message'>Next, identify the best-matching hexagon/rectangle for the input data (2015-07-16 20:25:00)...</strong>
<strong class='message'>Finally, append the response data (hits and mqe) into the sMap object (2015-07-16 20:25:00)...</strong>
<strong class='message'></strong>
<strong class='message'>Below are the summaries of the training results:</strong>
<strong class='message'>   dimension of input data: 18x18
   xy-dimension of map grid: xdim=9, ydim=6
   grid lattice: rect
   grid shape: sheet
   dimension of grid coord: 54x2
   initialisation method: linear
   dimension of codebook matrix: 54x18
   mean quantization error: 162.781134038408
</strong>
<strong class='message'>Below are the details of trainology:</strong>
<strong class='message'>   training algorithm: sequential
   alpha type: invert
   training neighborhood kernel: gaussian
   trainlength (x input data length): 30 at rough stage; 120 at finetune stage
   radius (at rough stage): from 2 to 1
   radius (at finetune stage): from 1 to 1
</strong>
<strong class='message'>End at 2015-07-16 20:25:00</strong>
<strong class='message'>Runtime in total is: 1 secs
</strong>
<div class='input'><a href='visCompReorder.html' target=visCompReorder>visCompReorder</a>(sMap, sReorder, title.rotate=15)
## As you have seen, reordered components of trained map is displayed. Each component illustrates a sample-specific map and is placed within a two-dimensional rectangular lattice. Across components/samples, genes with similar expression patterns are mapped onto the same position of the map. Geometric locations of components delineate relationships between components/samples, that is, samples with the similar expression profiles are placed closer to each other.
</div>
<a href="javascript:newWin('Fang-74.png', 'Fang-74.png', '1200', '1000')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='Fang-74.png', width="400" /></a>
      </pre>
    </div>

  <div class="span4">

    <h2>Source demo</h2>
    <a href="javascript:newWin('demo/Fang.r', 'demo/Fang.r', '1200', '1200')" title="Click to view"><B><code>Fang.r</code></B></a>
    
    <h2>Functions used in this demo</h2>
    <ul>
      <li><a href="sPipeline.html">sPipeline</a></li>
      <li><a href="visHexMulComp.html">visHexMulComp</a></li>
      <li><a href="sWriteData.html">sWriteData</a></li>
      <li><a href="visHexMapping.html">visHexMapping</a></li>
      <li><a href="visHexPattern.html">visHexPattern</a></li>
      <li><a href="sDmatCluster.html">sDmatCluster</a></li>
      <li><a href="visDmatCluster.html">visDmatCluster</a></li>
      <li><a href="visColormap.html">visColormap</a></li>
      <li><a href="visDmatHeatmap.html">visDmatHeatmap</a></li>
      <li><a href="sCompReorder.html">sCompReorder</a></li>
      <li><a href="visCompReorder.html">visCompReorder</a></li>
    </ul>
    
    <!--
    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) supraHex: an R/Bioconductor package for tabular omics data analysis using a supra-hexagonal map. <I>Biochemical and Biophysical Research Communications</I>, 443(1), 285-289. <a href="http://dx.doi.org/10.1016/j.bbrc.2013.11.103">doi:10.1016/j.bbrc.2013.11.103</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24309102">24309102</a></p>

    

    
    <h2>URL</h2>
    <ul>
      <li><a href="http://suprahex.r-forge.r-project.org">http://suprahex.r-forge.r-project.org</a></li>
    </ul>
    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    -->
    
  </div>

</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@well.ox.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>