<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>Home @ supraHex 1.5.1</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="Hai Fang and Julian Gough">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">supraHex 1.5.1</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="faqs.html"><i class="icon-question-sign icon-white"></i> FAQs</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

<script>
  (function() {
    var cx = '012094530736424854714:5dmiiel8qeq';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = (document.location.protocol == 'https:' ? 'https:' : 'http:') +
        '//www.google.com/cse/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>

<div><gcse:search></gcse:search></div>

    <div class="container">
      <header>
        
      </header>
      
      <div class="row">
  <div class="span8">
    <p><a href="faqs.html"><IMG src="supraHex_logo.png" height="150px" id="logo"></a></p>

<p><B><h4>An open-source R/Bioconductor package for tabular omics data analysis using <code>a supra-hexagonal map</code></h4></B></p>

<h2>News</h2>

<p>The artwork called <a href="Best_Artwork_Award_ISMB2014.pdf" target="artwork" style="font-size: 12px; color: #F87217; text-decoration: overline; border-bottom: 1px solid #F87217">supraHex</a> has won <a href="http://www.iscb.org/ismb2014-general-info/2234-ismb2014-award-winners#art" target="ISMB2014" >The Best Artwork Award in ISMB 2014</a>. This artwork is automatically done and is reproducible <a href="demo-ISMB2014.html" target="ISMB2014">here</a>.</p>

<h2>Introduction</h2>

<p><code>The supra-hexagonal map</code> is a giant hexagon on a 2-dimensional map grid seamlessly consisting of smaller hexagons. </p>

<p><code>supraHex</code> intends to meet the need for quickly understanding genome-wide biological data, which usually involve a large number of genomic coordinates (e.g. genes) but a much smaller number of samples. </p>

<p><code>supraHex</code> first uses a supra-hexagonal map to self-organise the input omics data, and then post-analyses the trained map for integrated tasks: simultaneous analysis of genes and samples, and multilayer omics data comparisons.</p>

<p><code>supraHex</code> aims to deliver an eye-intuitive tool and a dedicated website with extensive online documentation and easy-to-follow demos.</p>

<p>For more, see <a href="slides_supraHex.pdf" target="slides" style="font-size: 12px; color: #F87217; text-decoration: overline; border-bottom: 1px solid #F87217">slides</a> and <a href="poster_ISMB2014.png" target="slides" style="font-size: 12px; color: #F87217; text-decoration: overline; border-bottom: 1px solid #F87217">poster in ISMB2014</a>.</p>

<h2>Features</h2>

<ul>
<li>The supra-hexagonal map trained via a self-organising learning algorithm;</li>
<li>Visualisations at and across nodes of the map;</li>
<li>Partitioning of the map into gene meta-clusters;</li>
<li>Sample correlation on 2D sample landscape;</li>
<li>Overlaying additional data onto the trained map for exploring relationships between input and additional data;</li>
<li>Support for heatmap and tree building and visualisations;</li>
<li>Used by the package <a href="http://dnet.r-forge.r-project.org">dnet</a> for network-based sample classifications;</li>
<li>This package can run on <code>Windows</code>, <code>Mac</code> and <code>Linux</code>.</li>
</ul>

<h2>Workflow</h2>

<p><a href="javascript:newWin('supraHex_workflow.png', 'supraHex_workflow.png', '1200', '600')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='supraHex_workflow.png', width="800px" /></a></p>


  </div>
  
  <div class="span3 offset1">
  

    <h2>Manual</h2>
    <ul>
    
        <li>supraHex User Manual (<a href="vignettes/supraHex_vignettes.pdf">PDF</a>, <a href="vignettes/supraHex_vignettes.Rnw">Rnw</a>)</li>
    
    
    
        <li>supraHex Reference Manual (<a href="vignettes/supraHex.pdf">PDF</a>, <a href="docs.html">HTML</a>)</li>
    
    </ul>

    <h2>Demos</h2>
    <ul>
      <li><a href="demo-Hiratani.html">Demo for multilayer omics dataset from Hiratani et al</a></li>
      <li><a href="demo-Golub.html">Demo for leukemia patient dataset from Golub et al</a></li>
      <li><a href="demo-Fang.html">Demo for human embryo dataset from Fang et al</a></li>
      <li><a href="demo-Xiang.html">Demo for arabidopsis embryo dataset from Xiang et al</a></li>
      <li><a href="demo-ISMB2014.html">Demo for for the artwork in ISMB 2014</a></li>
    </ul>
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/supraHex">http://supfam.org/supraHex</a></li>
    </ul>
    
    <h2>Dependency</h2>
    <ul>
      <li><strong>Depends</strong>: hexbin</li>
      <li><strong>Imports</strong>: ape, MASS</li>
      <li><strong>Suggests</strong>: </li>
      <li><strong>Extends</strong>: </li>
    </ul>
    
    <h2>License</h2>
    GPL-2
    
    <h2>Author</h2>
    Hai Fang and Julian Gough
    
    <!--
    
    <h2>Vignette</h2>
    <ul>
        <li>supraHex User Manual (<a href="vignettes/supraHex_vignettes.pdf">PDF</a>, <a href="vignettes/supraHex_vignettes.Rnw">Rnw</a>)</li>
    </ul>
    
    <h2>Manual</h2>
    <ul>
        <li><a href="vignettes/supraHex.pdf">supraHex Reference Manual</a></li>
    </ul>
    
    <h2>Maintainer</h2>
    Hai Fang &lt;hfang@cs.bris.ac.uk&gt;
    
    -->
    
  </div>
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>