<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
               "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
lang="en" xml:lang="en">
<head>
<title>supraHex: An open-source package for tabular omics data analysis using a supra-hexagonal map</title>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8"/>
<meta name="generator" content="Org-mode"/>
<meta name="generated" content="2011-12-07 15:03:26 EST"/>
<meta name="author" content="Hai Fang"/>
<meta name="description" content=""/>
<meta name="keywords" content=""/>
<style type="text/css">
 <!--/*--><![CDATA[/*><!--*/
  html { font-family: Times, serif; font-size: 12pt; }
  .title  { text-align: center; }
  .todo   { color: red; }
  .done   { color: green; }
  .tag    { background-color: #add8e6; font-weight:normal }
  .target { }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  p.verse { margin-left: 3% }
  pre {
	border: 1pt solid #AEBDCC;
	background-color: #F3F5F7;
	padding: 5pt;
	font-family: courier, monospace;
        font-size: 90%;
        overflow:auto;
  }
  table { border-collapse: collapse; }
  td, th { vertical-align: top; }
  dt { font-weight: bold; }
  div.figure { padding: 0.5em; }
  div.figure p { text-align: center; }
  .linenr { font-size:smaller }
  .code-highlighted {background-color:#ffff00;}
  .org-info-js_info-navigation { border-style:none; }
  #org-info-js_console-label { font-size:10px; font-weight:bold;
                               white-space:nowrap; }
  .org-info-js_search-highlight {background-color:#ffff00; color:#000000;
                                 font-weight:bold; }
  /*]]>*/-->
</style>
<script type="text/javascript">
<!--/*--><![CDATA[/*><!--*/
 function CodeHighlightOn(elem, id)
 {
   var target = document.getElementById(id);
   if(null != target) {
     elem.cacheClassElem = elem.className;
     elem.cacheClassTarget = target.className;
     target.className = "code-highlighted";
     elem.className   = "code-highlighted";
   }
 }
 function CodeHighlightOff(elem, id)
 {
   var target = document.getElementById(id);
   if(elem.cacheClassElem)
     elem.className = elem.cacheClassElem;
   if(elem.cacheClassTarget)
     target.className = elem.cacheClassTarget;
 }
/*]]>*///-->
</script>
</head>
<body>
<div id="content">

<h1 class="title">supraHex: An open-source package for tabular omics data analysis using a supra-hexagonal map</h1>

<p align="left">
The package supraHex is developed to meet the need for quickly understanding genome-wide biological data, which usually involve a large number of genomic coordinates (e.g. genes) but a much smaller number of samples. It devises a novel map shape called <a href="#fig1">supra-hexagonal map</a> (a giant hexagon on a 2-dimensional (2D) map grid seamlessly consisting of smaller hexagons) to self-organise the input data. This trained map is then used for integrated tasks, highlighted by <a href="#fig2">simultaneous analysis of genes and samples</a>, and the overlaying of additional data (if any) onto the map for <a href="#fig3">multilayer omics data comparisons</a>.
</p>

<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#sec-1">1 Citation </a></li>
<li><a href="#sec-2">2 Availability </a></li>
<li><a href="#sec-3">3 Help </a></li>
<li><a href="#sec-4">4 Realcase </a></li>
</ul>
</div>
</div>

<div id="outline-container-1" class="outline-2">
    <h2 id="sec-1"><span class="section-number-2">1</span> Citations </h2>
    <div class="outline-text-2" id="text-1">
    <p>Fang H, Gough J. (2013) supraHex: an R/Bioconductor package for tabular omics data analysis using a supra-hexagonal map. <I>Biochemical and Biophysical Research Communications</I>, doi: <a href="http://dx.doi.org/10.1016/j.bbrc.2013.11.103" target="supraHex BBRC">http://dx.doi.org/10.1016/j.bbrc.2013.11.103</a>
    </p>
    </div>
</div>

<div id="outline-container-2" class="outline-2">
<h2 id="sec-2"><span class="section-number-2">2</span> Availability </h2>
<div class="outline-text-2" id="text-2">
    <p>Available at <a href="http://r-forge.r-project.org/projects/suprahex">http://r-forge.r-project.org/projects/suprahex</a> as part of <a href="https://r-forge.r-project.org">R-Forge</a>, and also available as part of <a href="http://bioconductor.org">Bioconductor</a> (see <a href="http://bioconductor.org/packages/release/bioc/html/supraHex.html">stable release version</a> and <a href="http://bioconductor.org/packages/devel/bioc/html/supraHex.html">latest development version</a>).
    </p>
    
    <p>Tips for the installation: </p>
    <LI>stable release version
<pre><code>
&gt; source("http://bioconductor.org/biocLite.R")
&gt; biocLite("supraHex")
</code></pre>
    <LI>latest development version (<b>highly recommended</b>).
<pre><code>
&gt; install.packages("hexbin",repos="http://www.stats.bris.ac.uk/R")
&gt; install.packages("supraHex",repos="http://R-Forge.R-project.org", type="source")
</code></pre>
    
    <p>More instructions on how to install are detailed in <a href="http://supfam.org/supraHex" target="supraHex tutorial">http://supfam.org/supraHex</a>.</p>
</div>
</div>


<div id="outline-container-3" class="outline-2">
    <h2 id="sec-3"><span class="section-number-2">3</span> Help </h2>
    <div class="outline-text-2" id="text-3">
    <p>There are 4 ways available to get you started:</p>
    <li> <a href="http://supfam.org/SUPERFAMILY/dcGO/supraHex/slides_supraHex.pdf" target="supraHex slides">Introductory slides</a> to give you overview of the concept and functionalities behind the package.
    <li> <a href="http://supfam.org/SUPERFAMILY/dcGO/supraHex/supraHex.Rcheck/supraHex/doc/supraHex_vignettes.pdf" target="supraHex user">User manual (vignette)</a> to provide a task-oriented description of the package functionality.
    <li> <a href="http://supfam.org/SUPERFAMILY/dcGO/supraHex/supraHex.Rcheck/supraHex-manual.pdf" target="supraHex reference">Reference manual</a> to provide a greater detail of specifications of functions. For easy reference online, this <a href="http://supfam.org/SUPERFAMILY/dcGO/supraHex/supraHex.Rcheck/supraHex/html/00Index.html" target="supraHex help">Online HTML Documentation</a> is also provided.
    <li> Many step-by-step tutorials on featuring realcases can be found in <a href="http://supfam.org/supraHex" target="supraHex tutorial">http://supfam.org/supraHex</a>.
    </div>
</div>


<div id="outline-container-4" class="outline-2">
<h2 id="sec-4"><span class="section-number-2">4</span> Realcase </h2>
<div class="outline-text-2" id="text-4">

<p>
To showcase the multifaceted functionalities accomplished by supraHex, we used published multilayer omics datasets on DNA replication timing, promoter CpG classification and gene expression (see <a href="#fig1">Figure 1</a>). These datasets consist of digitised replication timing, promoter CpG status and expression levels of 17,292 genes in a variety of samples. The DNA replication timing data matrix is first used as the input for training, producing a replication timing map (see <a href="#fig2">Figure 2</a>). Then, additional CpG and expression data are overlaid onto the trained replication-timing map, revealing an inherent relationship between late replication, low CpG density promoters and low expression levels (see <a href="#fig3">Figure 3</a>). Full descriptions can be found in <a href="http://dx.doi.org/10.1016/j.bbrc.2013.11.103" target="supraHex BBRC">supraHex publication</a>.

</p>

<table class="image">
    <tr>
    <img src="images/Fig1_suprahex_BBRC.png" height="600px" style="padding:30px 30px 5px 5px" id="fig1"/>
    </tr>
    <tr>
    <p align="left" style="font-family:arial;font-size:16px;"><b>Figure 1. The key functionalities in supraHex</b>. <br>(A) Architectural design of a supra-hexagonal map with node numbering. <br>(B) Illustration of datasets used to demonstrate the functionalities of supraHex. <br>(C) Schematic flowchart of the supraHex workflow. </p>
    </tr>
</table>

<table class="image">
    <tr>
    <img src="images/Fig2_suprahex_BBRC.png" height="600px" style="padding:30px 30px 5px 5px" id="fig2"/>
    </tr>
    <tr>
    <p align="left" style="font-family:arial;font-size:16px;"><b>Figure 2. The replication timing map of 22 cell types from mouse embryogenesis</b>. <br>(i) Each map illustrates a sample-specific replication-timing profile. <br>(ii) Across components/samples, genes with similar replication-timing patterns are mapped onto the same position in the map. <br>(iii) The outermost frame represents the replication-timing landscape for the samples analysed, the closer the more similar. </p>
    </tr>
</table>

<table class="image">
    <tr>
    <img src="images/Fig3_suprahex_BBRC.png" height="600px" style="padding:30px 30px 5px 5px" id="fig3"/>
    </tr>
    <tr>
    <p align="left" style="font-family:arial;font-size:16px;"><b>Figure 3. Using the replication-timing map to intuitively explore relationships between replication timing, CpG and expression</b>. <br>(A) CpG data overlaid onto the replication-timing map. The colour bar indicates the probability of containing genes with HCPs (left panel) or LCPs (right panel); the stronger, the higher. <br>(B) Expression data overlaid onto the replication-timing map. Each map illustrates a sample-specific expression profile. Here genes with similar replication-timing patterns are still mapped onto the same position on the map, but the illustrations are replaced with expression patterns.. </p>
    </tr>
</table>

</div>
</div>


</div>
</body>
</html>
