<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>visDmatHeatmap @ supraHex 1.1.17</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">supraHex 1.1.17</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to visualise gene clusters/bases partitioned from a supra-hexagonal grid using heatmap</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>visDmatHeatmap</code> is supposed to visualise gene
clusters/bases partitioned from a supra-hexagonal grid
using heatmap</p>
  
    <h3>Usage</h3>
    <pre>visDmatHeatmap(sMap, data, sBase, base.color = "rainbow", base.separated.arg = NULL, 
  base.legend.location = c("none", "bottomleft", "bottomright", "bottom", "left", 
          "topleft", "top", "topright", "right", "center"), reorderRow = c("none", 
          "hclust", "svd"), keep.data = F, ...)</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>sMap</dt>
      <dd>an object of class "sMap" or a codebook
matrix</dd>
      <dt>data</dt>
      <dd>a data frame or matrix of input data</dd>
      <dt>sBase</dt>
      <dd>an object of class "sBase"</dd>
      <dt>base.color</dt>
      <dd>short name for the colormap used to
encode bases (in row side bar). It can be one of "jet"
(jet colormap), "bwr" (blue-white-red colormap), "gbr"
(green-black-red colormap), "wyr" (white-yellow-red
colormap), "br" (black-red colormap), "yr" (yellow-red
colormap), "wb" (white-black colormap), and "rainbow"
(rainbow colormap, that is,
red-yellow-green-cyan-blue-magenta). Alternatively, any
hyphen-separated HTML color names, e.g.
"blue-black-yellow", "royalblue-white-sandybrown",
"darkgreen-white-darkviolet". A list of standard color
names can be found in
<a href = 'http://html-color-codes.info/color-names'>http://html-color-codes.info/color-names</a></dd>
      <dt>base.separated.arg</dt>
      <dd>a list of main parameters used
for styling bar separated lines. See 'Note' below for
details on the parameters</dd>
      <dt>base.legend.location</dt>
      <dd>location of legend to
describe bases. If "none", this legend will not be
displayed</dd>
      <dt>reorderRow</dt>
      <dd>the way to reorder the rows within a
base. It can be "none" for rows within a base being
reorded by the hexagon indexes, "hclust" for rows within
a base being reorded according to hierarchical clustering
of patterns seen, "svd" for rows within a base being
reorded according to svd of patterns seen</dd>
      <dt>keep.data</dt>
      <dd>logical to indicate whether or not to
also write out the input data. By default, it sets to
false for not keeping it. It is highly expensive to keep
the large data sets</dd>
      <dt>...</dt>
      <dd>additional graphic parameters used in
"visHeatmapAdv". For most parameters, please refer to
<a href = 'http://www.inside-r.org/packages/cran/gplots/docs/heatmap.2'>http://www.inside-r.org/packages/cran/gplots/docs/heatmap.2</a></dd>
    </dl>
    <!-- 
    -->
    
    <div class="Value">
      <h3>Value</h3>
      
      <p>a data frame with following components: <ul>
<li><code>ID</code>: ID for data. It inherits the rownames of
data (if exists). Otherwise, it is sequential integer
values starting with 1 and ending with dlen, the total
number of rows of the input data
</li>
<li><code>Hexagon_index</code>: the index for best-matching
hexagons </li>
<li><code>Cluster_base</code>: optional, it is only
appended when sBase is given. It stores the cluster
memberships/bases </li>
<li><code>data</code>: optional, it is only
appended when keep.data is true </li>
</ul>
 Note: the returned data
has rows in the same order as visualised in the heatmap</p>
    </div>
    <div class="Note">
      <h3>Note</h3>
      
      <p>A list of parameters in "base.separated.arg": <ul>
<li>"lty": the line type. Line types can either be
specified as an integer (0=blank, 1=solid (default),
2=dashed, 3=dotted, 4=dotdash, 5=longdash, 6=twodash) or as
one of the character strings
"blank","solid","dashed","dotted","dotdash","longdash","twodash",
where "blank" uses 'invisible lines' (i.e., does not draw
them) </li>
<li>"lwd": the line width </li>
<li>"col": the line
color </li>
</ul></p>
      <p></p>
    </div>
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) generate an iid normal random matrix of 100x10
data <- matrix( rnorm(100*10,mean=0,sd=1), nrow=100, ncol=10)

# 2) get trained using by default setup
sMap <- sPipeline(data=data)
</div>
<strong class='message'>Start at 2014-04-08 10:31:43</strong>
<strong class='message'></strong>
<strong class='message'>First, define topology of a map grid...</strong>
<strong class='message'>Second, initialise the codebook matrix given a topology and input data...</strong>
<strong class='message'>Third, get training at the rough stage...</strong>
<strong class='message'>Fourth, get training at the finetune stage...</strong>
<strong class='message'>Next, identify the best-matching hexagon/rectangle for the input data...</strong>
<strong class='message'>Finally, append the response data (hits and mqe) into the sMap object...</strong>
<strong class='message'></strong>
<strong class='message'>Below are the summaries of the training results:</strong>
<strong class='message'>   dimension of input data: 100x10
   xy-dimension of map grid: xdim=9, ydim=9
   grid lattice: hexa
   grid shape: suprahex
   dimension of grid coord: 61x2
   initialisation method: linear
   dimension of codebook matrix: 61x10
   mean quantization error: 4.85941466034151
</strong>
<strong class='message'>Below are the details of trainology:</strong>
<strong class='message'>   training algorithm: batch
   alpha type: invert
   training neighborhood kernel: gaussian
   trainlength (x input data length): 7 at rough stage; 25 at finetune stage
   radius (at rough stage): from 3 to 1
   radius (at finetune stage): from 1 to 1
</strong>
<strong class='message'>End at 2014-04-08 10:31:43</strong>
<strong class='message'>Runtime in total is: 0 secs
</strong>
<div class='input'>
# 3) partition the grid map into clusters using region-growing algorithm
sBase <- sDmatCluster(sMap=sMap, which_neigh=1,
distMeasure="median", clusterLinkage="average")

# 4) heatmap visualisation
output <- visDmatHeatmap(sMap, data, sBase,
base.legend.location="bottomleft", labRow=NA)
</div>
<a href="javascript:newWin('visDmatHeatmap-18.png', 'visDmatHeatmap-18.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='visDmatHeatmap-18.png', width="400" /></a></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>visDmatHeatmap</li>
    </ul>
    <ul>
      
    </ul> 
    -->
      
    <h2>See also</h2>
    
<code><a href='sDmatCluster.html'>sDmatCluster</a></code>, <code><a href='visHeatmapAdv.html'>visHeatmapAdv</a></code>

    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) <a href="http://mail.elsevier-alerts.com/go.asp?/bESJ001/mIMVC2EF/qI4ML2EF/u2ZWFR3F/xR1CF2EF/cutf%2D8">supraHex: an R/Bioconductor package for tabular omics data analysis using a supra-hexagonal map</a>. Biochemical and Biophysical Research Communications, 443(1), 285-289. <a href="http://dx.doi.org/10.1016/j.bbrc.2013.11.103">http://dx.doi.org/10.1016/j.bbrc.2013.11.103</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24309102">24309102</a></p>

    
    <!--
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/supraHex">http://supfam.org/supraHex</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>