<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
<title>visHeatmap @ supraHex 1.2.1</title>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="author" content="">

<link href="css/bootstrap.css" rel="stylesheet">
<link href="css/bootstrap-responsive.css" rel="stylesheet">
<link href="css/highlight.css" rel="stylesheet">
<link href="css/staticdocs.css" rel="stylesheet">


<style type="text/css">

.navbar .brand {
  display: block;
  float: left;
  padding: 8px 20px 12px;
  margin-left: -20px;
  font-size: 20px;
  font-weight: 200;
  line-height: 1;
  color: #999999;
}

.navbar .brand:hover {
  color: #F87217;
  text-decoration: none;
  background-color: transparent;
}

.navbar .nav > li > a:hover {
  color: #F87217;
  text-decoration: underline;
  background-color: transparent;
}

</style>


<!--[if lt IE 9]>
  <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
<![endif]-->

<!-- google analytics -->
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-17338744-2']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

<!-- Enlarge image -->
<script type="text/javascript">
    function newWin(url, name, width, height) { 
       
        var left = (screen.width/2)-(width/2);
        var top = (screen.height/2)-(height/2); 
        
        window.open(url,name,'scrollbars=yes,resizable=yes,fullscreen=yes, width=' + width + ', height='+height+', top='+top+', left='+left);
    }
</script>
  </head>

  <body>
    <div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="brand" href="index.html">supraHex 1.2.1</a>
      <div class="nav">
        <ul class="nav">
          <li><a href="index.html"><i class="icon-home icon-white"></i> Home</a></li>
        </ul>
        <ul class="nav">
          <li><a href="docs.html"><i class="icon-book icon-white"></i> Documentations</a></li>
        </ul>
        <ul class="nav">
          <li><a href="install.html"><i class="icon-wrench icon-white"></i> Installation</a></li>
        </ul>
        <ul class="nav">
          <li><a href="demos.html"><i class="icon-play icon-white"></i> Demos</a></li>
        </ul>
        <ul class="nav">
          <li><a href="cite.html"><i class="icon-pencil icon-white"></i> Citation</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>

    <div class="container">
      <header>
        
      </header>
      
      <h2>Function to visualise input data matrix using heatmap</h2>

<div class="row">
  <div class="span8">
  
    <h3>Description</h3>
    <p><code>visHeatmap</code> is supposed to visualise input data
matrix using heatmap. Note: this heatmap displays matrix in
a bottom-to-top direction</p>
  
    <h3>Usage</h3>
    <pre>visHeatmap(data, scale = c("none", "row", "column"), row.metric = c("none", "pearson", 
  "spearman", "kendall", "euclidean", "manhattan", "cos", "mi"), row.method = c("ward", 
      "single", "complete", "average", "mcquitty", "median", "centroid"), column.metric = c("none", 
      "pearson", "spearman", "kendall", "euclidean", "manhattan", "cos", "mi"), column.method = c("ward", 
      "single", "complete", "average", "mcquitty", "median", "centroid"), colormap = c("bwr", 
      "jet", "gbr", "wyr", "br", "yr", "rainbow", "wb"), ncolors = 64, zlim = NULL, 
      row.cutree = NULL, row.colormap = c("rainbow"), column.cutree = NULL, column.colormap = c("rainbow"), 
      ...)</pre>
    
    <h3>Arguments</h3>
    <dl>
      <dt>data</dt>
      <dd>an input gene-sample data matrix used for
heatmap</dd>
      <dt>scale</dt>
      <dd>a character indicating when the input matrix
should be centered and scaled. It can be one of "none"
(no scaling), "row" (being scaled in the row direction),
"column" (being scaled in the column direction)</dd>
      <dt>row.metric</dt>
      <dd>distance metric used to calculate the
distance metric between rows. It can be one of "none"
(i.e. no dendrogram between rows), "pearson", "spearman",
"kendall", "euclidean", "manhattan", "cos" and "mi". See
details at
<a href = 'http://suprahex.r-forge.r-project.org/sDistance.html'>http://suprahex.r-forge.r-project.org/sDistance.html</a></dd>
      <dt>row.method</dt>
      <dd>the agglomeration method used to
cluster rows. This should be one of "ward", "single",
"complete", "average", "mcquitty", "median" or
"centroid". See 'Note' below for details</dd>
      <dt>column.metric</dt>
      <dd>distance metric used to calculate
the distance metric between columns. It can be one of
"none" (i.e. no dendrogram between rows), "pearson",
"spearman", "kendall", "euclidean", "manhattan", "cos"
and "mi". See details at
<a href = 'http://suprahex.r-forge.r-project.org/sDistance.html'>http://suprahex.r-forge.r-project.org/sDistance.html</a></dd>
      <dt>column.method</dt>
      <dd>the agglomeration method used to
cluster columns. This should be one of "ward", "single",
"complete", "average", "mcquitty", "median" or
"centroid". See 'Note' below for details</dd>
      <dt>colormap</dt>
      <dd>short name for the colormap. It can be
one of "jet" (jet colormap), "bwr" (blue-white-red
colormap), "gbr" (green-black-red colormap), "wyr"
(white-yellow-red colormap), "br" (black-red colormap),
"yr" (yellow-red colormap), "wb" (white-black colormap),
and "rainbow" (rainbow colormap, that is,
red-yellow-green-cyan-blue-magenta). Alternatively, any
hyphen-separated HTML color names, e.g.
"blue-black-yellow", "royalblue-white-sandybrown",
"darkgreen-white-darkviolet". A list of standard color
names can be found in
<a href = 'http://html-color-codes.info/color-names'>http://html-color-codes.info/color-names</a></dd>
      <dt>ncolors</dt>
      <dd>the number of colors specified over the
colormap</dd>
      <dt>zlim</dt>
      <dd>the minimum and maximum z/patttern values for
which colors should be plotted, defaulting to the range
of the finite values of z. Each of the given colors will
be used to color an equispaced interval of this range.
The midpoints of the intervals cover the range, so that
values just outside the range will be plotted</dd>
      <dt>row.cutree</dt>
      <dd>an integer scalar specifying the
desired number of groups being cut from the row
dendrogram. Note, this optional is only enabled when the
row dengrogram is built</dd>
      <dt>row.colormap</dt>
      <dd>short name for the colormap to
color-code the row groups (i.e. sidebar colors used to
annotate the rows)</dd>
      <dt>column.cutree</dt>
      <dd>an integer scalar specifying the
desired number of groups being cut from the column
dendrogram. Note, this optional is only enabled when the
column dengrogram is built</dd>
      <dt>column.colormap</dt>
      <dd>short name for the colormap to
color-code the column groups (i.e. sidebar colors used to
annotate the columns)</dd>
      <dt>...</dt>
      <dd>additional graphic parameters. Type ?heatmap
for the complete list.</dd>
    </dl>
    <!-- 
    -->
    
    <div class="Value">
      <h3>Value</h3>
      
      <p>invisible</p>
    </div>
    <div class="Note">
      <h3>Note</h3>
      
      <p>The clustering methods are provided: <ul>
<li>"ward": Ward's minimum variance method aims at
finding compact, spherical clusters </li>
<li>"single": The
single linkage method (which is closely related to the
minimal spanning tree) adopts a 'friends of friends'
clustering strategy </li>
<li>"complete": The complete linkage
method finds similar clusters
</li>
<li>"average","mcquitty","median","centroid": These
methods can be regarded as aiming for clusters with
characteristics somewhere between the single and complete
link methods. Two methods "median" and "centroid" are not
leading to a monotone distance measure, or equivalently the
resulting dendrograms can have so called inversions (which
are hard to interpret) </li>
</ul></p>
      <p></p>
    </div>
    
    <h3 id="examples">Examples</h3>
    <pre class="examples"><div class='input'># 1) generate data with an iid matrix of 100 x 9
data <- cbind(matrix(rnorm(100*3,mean=0,sd=1), nrow=100, ncol=3),
matrix(rnorm(100*3,mean=0.5,sd=1), nrow=100, ncol=3),
matrix(rnorm(100*3,mean=-0.5,sd=1), nrow=100, ncol=3))
colnames(data) <- c("S1","S1","S1","S2","S2","S2","S3","S3","S3")

# 2) prepare colors for the column sidebar
lvs <- unique(colnames(data))
lvs_color <- visColormap(colormap="rainbow")(length(lvs))
my_ColSideColors <- sapply(colnames(data), function(x)
lvs_color[x==lvs])

# 3) heatmap with row dendrogram (with 10 color-coded groups)
visHeatmap(data, row.metric="euclidean", row.method="average",
colormap="gbr", zlim=c(-2,2),
ColSideColors=my_ColSideColors, row.cutree=10, row.colormap="jet",
labRow=NA)
</div>
<a href="javascript:newWin('visHeatmap-2.png', 'visHeatmap-2.png', '1200', '1200')" title="Click to enlarge"><img style="max-width:95%;border:1px solid #EEEEEE;box-shadow:5px 5px 2px #C0C0C0;" src='visHeatmap-2.png', width="400" /></a></pre>
  </div>
  
  <div class="span4">
    
    <!-- 
    <ul>
      <li>visHeatmap</li>
    </ul>
    <ul>
      
    </ul> 
    -->
      
    <h2>See also</h2>
    
<code><a href='visHeatmap.html'>visHeatmap</a></code>

    
    <h2>Citation</h2>
    <p>Fang H, Gough J. (2014) <a href="http://mail.elsevier-alerts.com/go.asp?/bESJ001/mIMVC2EF/qI4ML2EF/u2ZWFR3F/xR1CF2EF/cutf%2D8">supraHex: an R/Bioconductor package for tabular omics data analysis using a supra-hexagonal map</a>. Biochemical and Biophysical Research Communications, 443(1), 285-289. <a href="http://dx.doi.org/10.1016/j.bbrc.2013.11.103">http://dx.doi.org/10.1016/j.bbrc.2013.11.103</a>, PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/?term=24309102">24309102</a></p>

    
    <!--
    
    <h2>URL</h2>
    <ul>
      <li><a href="http://supfam.org/supraHex">http://supfam.org/supraHex</a></li>
    </ul>
    
    
    -->
    
  </div>
  
</div>
      
      <footer>
      <p class="pull-right"><a href="#">Back to top</a></p>

<p><a href="mailto:hfang@cs.bris.ac.uk" title="Hai Fang">Hai Fang</a></p>
<p><a href="http://bioinformatics.bris.ac.uk">Computational Genomics Group, Department of Computer Science, University of Bristol, UK</a></p>


      </footer>
    </div>
  </body>
</html>